File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/news/NewsRepository.kt
package com.reguerta.user.domain.news

import com.reguerta.user.domain.access.Member

interface NewsRepository {
    suspend fun getNewsFor(member: Member): List<NewsArticle>

    suspend fun upsertNews(article: NewsArticle): NewsArticle

    suspend fun deleteNews(newsId: String): Boolean
}

File: ios/Reguerta/Reguerta/Domain/News/NewsRepository.swift
import Foundation

protocol NewsRepository: Sendable {
    func news(visibleTo member: Member, environment: SessionEnvironment) async throws -> [NewsArticle]
    func allNews(environment: SessionEnvironment) async throws -> [NewsArticle]
    func upsert(article: NewsArticle, environment: SessionEnvironment) async throws -> NewsArticle
    func delete(newsId: String, environment: SessionEnvironment) async throws -> Bool
}
